src/compas/scene/sceneobject.py

```diff
@@ -22,7 +21,50 @@
 from .descriptors.protocol import DescriptorProtocol
 
 
-class SceneObject(TreeNode):
+class SceneObjectFactory:
+    """Factory class for creating appropriate SceneObject instances based on input item type.
+
+    This factory encapsulates the logic for selecting the right SceneObject subclass
+    for a given data item, making the creation process more explicit and easier to understand.
+    """
+
+    @staticmethod
+    def create(item=None, scene=None, **kwargs):
```

Comment:
as mentioned, perhaps it is overkill to use a class for this, since there is only 1 static method...